kore/src/Kore/Internal/Substitution.hs

```diff
@@ -171,6 +172,28 @@ retractAssignment (PredicateEquals term (Var_ var)) =
     Just (assign var term)
 retractAssignment _ = Nothing
 
+{- | Extract an 'Assignment' for a /particular/ variable.
+
+Returns 'Nothing' if the 'Predicate' is not in the correct form or the variable
+name does not match.
+
+See alse: 'retractAssignment'
+-}
+retractAssignmentFor ::
+    InternalVariable variable =>
+    SomeVariableName variable ->
+    Predicate variable ->
+    Maybe (Assignment variable)
+retractAssignmentFor someVariableName predicate =
+    case predicate of
+        PredicateEquals (Var_ var) term
+            | variableName var == someVariableName ->
+                Just (Assignment_ var term)
+        PredicateEquals term (Var_ var)
+            | variableName var == someVariableName ->
+                Just (Assignment_ var term)
+        _ -> Nothing
```

Comment:
I'm not sure this is correct, if the `term` is a variable `var'`, and `var' < var`, with the code here I think we'd break the `Assignment` invariant